lib/private/User/User.php

```diff
@@ -319,6 +320,25 @@ public function setPassword($password, $recoveryPassword = null) {
 		}
 	}
 
+	public function getPasswordHash(): ?string {
+		if (!($this->backend instanceof IPasswordHashBackend)) {
+			return null;
+		}
+		/** @var IPasswordHashBackend $backend */
+		$backend = $this->backend;
+		return $backend->getPasswordHash($this->uid);
```

Comment:
```suggestion
		return $this->backend->getPasswordHash($this->uid);
```
With the test above psalm should already know the interface is implemented, no?